src/hypercall.rs

```diff
@@ -183,42 +268,53 @@ pub fn read(
 	}
 }
 
+/// Handles a v1 write hypercall (for which a guest-provided guest virtual address must be
+/// converted to a guest physical address by the host).
+pub fn write_v1(
+	peripherals: &VmPeripherals,
+	syswrite: &v1::parameters::WriteParams,
+	root_pt: GuestPhysAddr,
+	file_map: &mut UhyveFileMap,
+) -> io::Result<()> {
+	if let Ok(guest_phys_addr) = virt_to_phys(syswrite.buf, &peripherals.mem, root_pt) {
+		let tmp = v2::parameters::WriteParams {
+			fd: syswrite.fd,
+			buf: guest_phys_addr,
+			len: syswrite.len,
+		};
+		write(peripherals, &tmp, file_map)
+	} else {
+		warn!("Unable to convert guest virtual address into guest physical address");
+		Ok(())
+	}
+}
 /// Handles an write syscall on the host.
 pub fn write(
 	peripherals: &VmPeripherals,
-	syswrite: &WriteParams,
-	root_pt: GuestPhysAddr,
+	syswrite: &v2::parameters::WriteParams,
 	file_map: &mut UhyveFileMap,
 ) -> io::Result<()> {
 	let mut bytes_written: usize = 0;
 	while bytes_written != syswrite.len {
-		let guest_phys_addr = virt_to_phys(
-			syswrite.buf + bytes_written as u64,
-			&peripherals.mem,
-			root_pt,
-		);
+		let guest_phys_addr = syswrite.buf + bytes_written as u64;
 
-		if let Ok(guest_phys_addr) = guest_phys_addr {
-			if syswrite.fd == 1 || syswrite.fd == 2 {
-				let bytes = unsafe {
-					peripherals
-						.mem
-						.slice_at(guest_phys_addr, syswrite.len)
-						.map_err(|e| {
-							io::Error::new(
-								io::ErrorKind::InvalidInput,
-								format!("invalid syswrite buffer: {e:?}"),
-							)
-						})?
-				};
-				return peripherals.serial.output(bytes);
-			} else if !file_map.fdmap.is_fd_present(syswrite.fd.into_raw_fd()) {
-				// We don't write anything if the file descriptor is not available,
-				// but this is OK for now, as we have no means of returning an error code
-				// and writes are not necessarily guaranteed to write anything.
-				return Ok(());
-			}
-		} else {
+		if syswrite.fd == 1 || syswrite.fd == 2 {
+			let bytes = unsafe {
+				peripherals
+					.mem
+					.slice_at(guest_phys_addr, syswrite.len)
+					.map_err(|e| {
+						io::Error::new(
+							io::ErrorKind::InvalidInput,
+							format!("invalid syswrite buffer: {e:?}"),
+						)
+					})?
+			};
+			return peripherals.serial.output(bytes);
+		} else if !file_map.fdmap.is_fd_present(syswrite.fd.into_raw_fd()) {
+			// We don't write anything if the file descriptor is not available,
+			// but this is OK for now, as we have no means of returning an error code
```

Comment:
Given that this is a new interface version that isn't currently deployed, wouldn't it make sense to adjust the write hypercall params to also have a way of returning non-fatal errors?